### PR TITLE
Add compatibility layer documentation

### PR DESCRIPTION
**Pull request by gigitux**
This PR adds the documentation for the compatibility layer for the blockified templates.

### CODE REVIEW

File: docs/internal-developers/blockified-templates/compatibility-layer.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Compatibility Layer - [AbstractTemplateCompatibility.php](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/Templates/AbstractTemplateCompatibility.php)
+
+The Compatibility Layer ensures that blockified templates work correctly with extensions that use hooks to extend their behavior. It appends/pre-appends the corresponding hooks to each block. Also, it removes the default callbacks added to those hooks by WooCommerce.
+The Compatibility Layer disables when the `WooCommerce Single Product`, `WooCommerce Product Attribute`, `WooCommerce Product Taxonomy`, `WooCommerce Product Tag`, `WooCommerce Product Search Results` or `WooCommerce Product Grid` block is added on the page. Furthermore, it is possible disable the compatibility layer via the hook: [`woocommerce_disable_compatibility_layer`](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/Templates/AbstractTemplateCompatibility.php/#L41-L42).
```
Comment (kmanijak) [inline docs/internal-developers/blockified-templates/compatibility-layer.md]:
```suggestion
The Compatibility Layer disables when the `WooCommerce Single Product`, `WooCommerce Product Attribute`, `WooCommerce Product Taxonomy`, `WooCommerce Product Tag`, `WooCommerce Product Search Results` or `WooCommerce Product Grid` block is added on the page. Furthermore, it is possible to disable the compatibility layer via the hook: [`woocommerce_disable_compatibility_layer`](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/Templates/AbstractTemplateCompatibility.php/#L41-L42).
```

**Review by kmanijak — APPROVED** [review]
I found one typo, except for that - it looks great! Thanks for covering Product Archive Template as well 💪

**Review by albarin — APPROVED** [review]
🙇‍♀️

**Comment by gigitux** [comment]
Thanks for the feedback! I will merge it!

**Pull request merged**